nautobot/extras/tests/test_datasources.py

```diff
@@ -275,19 +279,24 @@ def create_empty_repo(path, url):
                 MockGitRepo.return_value.checkout.return_value = self.COMMIT_HEXSHA
 
                 # Run the Git operation and refresh the object from the DB
-                pull_git_repository_and_refresh_data(self.repo.pk, self.mock_request, self.job_result.pk)
-                self.job_result.refresh_from_db()
+                # pull_git_repository_and_refresh_data(self.repo.pk, self.mock_request, self.job_result.pk)
+                job_result = create_job_result_and_run_job(
+                    module="nautobot.core.jobs",
+                    name="GitRepositorySync",
+                    source="system",
+                    repository=self.repo.pk,
+                )
 
                 self.assertEqual(
-                    self.job_result.status,
+                    job_result.status,
                     JobResultStatusChoices.STATUS_SUCCESS,
-                    self.job_result.data,
+                    job_result.result,
```

Comment:
the `run` method of this job doesn't return anything so shouldn't `job_result.result` be `None`?

Reply:
Might be better to use `job_result.traceback`?